### PR TITLE
Initial documentation for Java web app using Tomcat guide

### PR DESCRIPTION
**Pull request by davidwrede**
Guide and getting started pages are in a rough draft form that can be built upon.

Signed-off-by: David Wrede <dwrede@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by mwrock** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/318230296).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/318230296).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

File: www/source/guides/tomcat/index.html.slim [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+     )
+  p
+   | For more information on the variables you can set in your plan, see #{link_to 'Plan Settings','/docs/reference/#plan-settings'}.
+hr
+section#logic.get-started--boxes
+  h3.h3 Define Build Logic
+  p.description
+    | When building your Java web application, several build tools are available. In this example, we chose Maven to build our <code>.war</code> file. The appropriate place to do this build operation should be done in the <code>do_build</code> callback.
+  p
+    | After your package is built, it needs to be installed into the correct directory in your Habitat package. To do this, use the <code>do_install</code> callback and copy the <code>.war</code> file and place it in the root of the Habitat package that you are building.
```
Comment (mwrock) [inline www/source/guides/tomcat/index.html.slim]: `and place it in` -> `to`

File: www/source/guides/tomcat/index.html.slim [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  p
+   | For more information on build callbacks and the variables you can use within them, see #{link_to 'Build Phase Callbacks','/docs/reference/#reference-callbacks'} and #{link_to 'Plan Variables','/docs/reference/#plan-variables'}.
+hr
+section#hooks.get-started--boxes
+  h3.h3 Define Hooks
+  p.description
+    | Your Java web app will not run correctly unless some additional steps are taken at runtime. Application lifecycle hooks are available for you to define this logic during application initialization, while your application is running, when new configuration changes are applied, etc.
+  
+  p
+   | Before your application can start, you need to set up the runtime environment for your application. Initialization operations like this are typically done in an <code>init</code> hook. In this case, because we're using Tomcat, we will copy our Tomcat directory from our package into the <code>var</code> directory that our service will use during runtime.
```
Comment (mwrock) [inline www/source/guides/tomcat/index.html.slim]: Might be more clear if the `Init` heading was above this `p`

File: www/source/guides/tomcat/index.html.slim [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+     exec 2>&1
+
+     echo "Preparing TOMCAT_HOME..."
+
+     # Create a Tomcat root for this app in the package's service directory
+     cp -a {{pkgPathFor "core/tomcat8"}}/tc {{pkg.svc_var_path}}/
+
+     echo "Done preparing TOMCAT_HOME"
+
+  p
```
Comment (mwrock) [inline www/source/guides/tomcat/index.html.slim]: Same comment as above but with the `Run` heading

**Comment by ryankeairns** [comment]
I asked Dave to jot this info down as he transitioned out in order to a) capture some of his knowledge around the topic and b) demonstrate how we could structure a new language-based guide. I'll close it without merging then we can simply refer back to it once we begin adding more of these in the future.

**Pull request closed without merging**